sonar-scm-git-plugin/src/test/java/org/sonarsource/scm/git/GitScmProviderTest.java

```diff
@@ -152,13 +152,13 @@ public void branchChangedFiles_from_merged_and_diverged() throws IOException, Gi
 
   @Test
   public void branchChangedFiles_when_git_work_tree_is_above_project_basedir() throws IOException, GitAPIException {
-    Path projectDir = worktree.resolve("project");
-    Files.createDirectory(projectDir);
-
     git.branchCreate().setName("b1").call();
     git.checkout().setName("b1").call();
+    
+    Path projectDir = worktree.resolve("project");
+    Files.createDirectory(projectDir);
     createAndCommitFile(projectDir, "file-b1");
-
+    commit("project");
```

Comment:
Is this necessary?

Reply:
yes, IIRC otherwise the file in that directory is not committed. I will add another test to ensure that uncomitted files are now not included in the diff.

Reply:
I don't see anything else that needs to be committed here, except for the "project/file-b1", which is already committed in `createAndCommitFile`.

Nice idea for the test!

Reply:
file-b1 doesn't actually get committed, probably another limitation of JGit.